From c7c18bc93441872e7ff2fa47a18a12ef319b7949 Mon Sep 17 00:00:00 2001
From: Franklin S. Cooper Jr <fcooper@ti.com>
Date: Sat, 6 Oct 2012 12:32:29 -0500
Subject: [PATCH] iperf: Fix CPU Usage

* Updated the patch 000-Iperf_Fix-CPU-Usage.diff to apply to iperf v2.0.5
* That patch was largely based on the patch:
  http://marc.info/?l=linux-kernel&m=119088670113210&w=2
* The majority of the changes from 000-Iperf_Fix-CPU-Usage.diff have been
  incorporated into iperf v2.0.5.
* The two remaining thread_rest changes that 000-Iperf_Fix-CPU-Usage.diff
  made have been added to this version of iperf v2.0.5

Upstream-Status: Inappropriate [not author]

Signed-off-by: Franklin S. Cooper Jr <fcooper@ti.com>
---
 src/Reporter.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Reporter.c b/src/Reporter.c
index 7d9263b..4fbb712 100644
--- a/src/Reporter.c
+++ b/src/Reporter.c
@@ -391,7 +391,7 @@ void EndReport( ReportHeader *agent ) {
     if ( agent != NULL ) {
         int index = agent->reporterindex;
         while ( index != -1 ) {
-            thread_rest();
+            Condition_Wait( &ReportDoneCond );
             index = agent->reporterindex;
         }
         agent->agentindex = -1;
@@ -412,7 +412,7 @@ void EndReport( ReportHeader *agent ) {
 Transfer_Info *GetReport( ReportHeader *agent ) {
     int index = agent->reporterindex;
     while ( index != -1 ) {
-        thread_rest();
+        Condition_Wait( &ReportDoneCond );
         index = agent->reporterindex;
     }
     return &agent->report.info;
--
1.7.0.4
